lib/sdf-server/src/server/service/v2/func.rs

```diff
@@ -0,0 +1,170 @@
+                StatusCode::FORBIDDEN
+            }
+            // these errors represent problems with the shape of the request
+            Self::MissingActionKindForActionFunc
+            | Self::MissingActionPrototype
+            | Self::MissingFuncId
+            | Self::MissingInputLocationForAttributeFunc
+            | Self::MissingOutputLocationForAttributeFunc
+            | Self::MissingPrototypeId
+            | Self::MissingSchemaVariantAndFunc => StatusCode::BAD_REQUEST,
+            // When a graph node cannot be found for a schema variant, it is not found
+            Self::SchemaVariant(dal::SchemaVariantError::NotFound(_)) => StatusCode::NOT_FOUND,
+            _ => ApiError::DEFAULT_ERROR_STATUS_CODE,
+        };
+
+        ApiError::new(status_code, self).into_response()
+    }
+}
+
+pub fn v2_routes() -> Router<AppState> {
+    Router::new()
+        // Func Stuff
+        .route("/", get(list_funcs::list_funcs))
+        .route("/code", get(get_code::get_code)) // accepts a list of func_ids
+        .route("/create", post(create_func::create_func))
+        .route("/:func_id/update", post(update_func::update_func)) // only save the func's metadata
+        .route("/:func_id/save_code", post(save_code::save_code)) // only saves func code
+        .route("/:func_id/test_execute", post(test_execute::test_execute))
+        .route("/:func_id/execute", post(execute_func::execute_func))
+        // Func Bindings
+        .route(
+            "/:func_id/bindings/create",
+            post(binding::create_binding::create_binding),
+        )
+        .route(
+            "/:func_id/bindings/delete",
+            post(binding::delete_binding::delete_binding),
+        )
+        .route(
+            "/:func_id/bindings/update",
+            post(binding::update_binding::update_binding),
+        )
+        // Attribute Bindings
+        .route(
+            "/:func_id/create_attribute_binding",
+            post(binding::attribute::create_attribute_binding::create_attribute_binding),
+        )
+        .route(
+            "/:func_id/reset_attribute_binding",
+            post(binding::attribute::reset_attribute_binding::reset_attribute_binding),
+        )
+        .route(
+            "/:func_id/update_attribute_binding",
+            post(binding::attribute::update_attribute_binding::update_attribute_binding),
+        )
+        // Func Arguments
+        .route(
+            "/:func_id/:func_argument_id/update",
+            post(argument::update_argument::update_func_argument),
+        )
+        .route(
+            "/:func_id/create_argument",
+            post(argument::create_argument::create_func_argument),
+        )
+        .route(
+            "/:func_id/:func_argument_id/delete",
+            post(argument::delete_argument::delete_func_argument),
+        )
+}
+
+pub async fn get_code_response(ctx: &DalContext, func_id: FuncId) -> FuncAPIResult<FuncCode> {
```

Comment:
I'm not sure I get 100% why this is necessary, I think a comment over here would help